### PR TITLE
2021 Updates

### PR DESCRIPTION
**Pull request by daronyondem**

### CODE REVIEW

**Review by hkamel — CHANGES_REQUESTED** [review]

**Comment by rajanbhayana** [comment]
Please introduce a basic logic app and API Management example to this. Like Email notifications for logic app and adding some external API to APIM (like order status). Almost all customers looking at websites dig into logic apps and api management at some point. Thanks.

File: Hands-on lab/HOL step-by-step - App modernization.md [hunk 1]
```diff
@@ -29,1852 +29,1088 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
 
-## Requirements
-
-- Microsoft Azure subscription
-- A virtual machine configured with Visual Studio Community 2019 or higher (setup in the Before the hands-on lab exercises)
-- **IMPORTANT**: To complete this lab, you must have sufficient rights within your Azure AD tenant to:
-  - Create an Azure Active Directory application and service principal
-  - Assign roles on your subscription
-  - Register resource providers
+The solution begins with setting up Azure Migrate as the central assessment and migration hub for Parts Unlimited's E-Commerce web site. Using the App Service Migration tool from Azure Migrate, Parts Unlimited found that their web site is fully compatible with Azure App Service. As a next step, they use App Service Migration to provision an Azure App Service environment and deploy their application to Azure. Following the success in moving the web application, Parts Unlimited uses the Data Migration Assistant (DMA) assessment to determine that they can migrate into a fully-managed SQL Database service in Azure. The assessment reveals no compatibility issues or unsupported features that would prevent them from using Azure SQL Database.
```
Comment (johrose) [inline Hands-on lab/HOL step-by-step - App modernization.md:86]: "App Service Migration Assistant, can be shortened to "Migration Assistant" in the context of the app migration

File: Hands-on lab/HOL step-by-step - App modernization.md [hunk 2]
```diff
@@ -29,1852 +29,1088 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
+   ![AppServiceMigrationAssistant is highlighted on the desktop.](media/appservicemigrationassistant-desktop.png)
 
-7. Select **Connect**.
+9. Once App Service Migration Assistant discovers the web sites availabie on the server choose **Default Web Site (1)** for migration and select **Next (2)** to start assessment.
 
-8. On the **Add sources** dialog that appears next, check the box for `ContosoInsurance` and select **Add**.
+   ![AppServiceMigrationAssistant is open. Default Web Site is selected. Next button is highlighted.](media/appservicemigration-choose-site.png)
 
-   ![The ContosoInsurance box is checked on the Add sources dialog.](media/dma-add-sources.png "Add sources")
+20. Observe the result of the assessment report. In our case, our application has successfully passed 13 tests **(1)** with no additional actions needed. Now that our assessment is complete, select **Next (2)** to proceed to the migration.
```
Comment (johrose) [inline Hands-on lab/HOL step-by-step - App modernization.md:210]: Might be worth surfacing this resource here. It lists remediation recommendations if any of the compatibility checks fail : https://github.com/Azure/App-Service-Migration-Assistant/wiki/Readiness-Checks

File: Hands-on lab/HOL step-by-step - App modernization.md [hunk 3]
```diff
@@ -29,1852 +29,1088 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
 
-2. In the New project dialog, enter the following:
+    ![Azure Migrate Project is set to partsunlimitedweb. The next button is highlighted.](media/appservicemigration-azure-migrate.png)
 
-   - **Project type**: Select Migration.
-   - **Project name**: Enter Migration.
-   - **Source server type**: Select SQL Server.
-   - **Target server type**: Select Azure SQL Database.
-   - **Migration scope**: Select Schema only.
+6. In order to migrate Parts Unlimited web site we have to create an App Service Plan and an App Service. The Azure App Service Migration Assistant will take care of all the requirements needed. Select **Use existing (1)** and select the lab resource group as your deployment target. For the App Service in Azure we have to provude a site name that is globally unique. We suggest using a pattern that matches `partsunlimited-web-{uniquesuffix}` **(2)**. Select **Migrate** to start the migration process.
```
Comment (johrose) [inline Hands-on lab/HOL step-by-step - App modernization.md:238]: You can remove "and an App Service" from the first sentence. At this step we are selecting/choosing an appropriate App Service plan to power our Web App

Comment (johrose) [inline Hands-on lab/HOL step-by-step - App modernization.md:238]: Also would reframe this sentence: "For the App Service in Azure we have to provude a site name that is globally unique. We suggest using...." to "App Service requires a globally unique Site Name, we suggest using..."

File: Hands-on lab/HOL step-by-step - App modernization.md [hunk 4]
```diff
@@ -29,1852 +29,1088 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
+   ![Connection string panel if SQL Database is open. Copy button for ADO.NET connection string is highlighted.](media/sql-connection-string-copy.png)
 
-1. Return to the Overview blade of your SQL database in the [Azure portal](https://portal.azure.com), and select **Dynamic Data Masking** from the left-hand menu.
+3. Paste the value into a text editor, such as Notepad.exe, to replace the Password placeholder. Replace the `{your_password}` section with your admin password. Copy the full connection string with the replaced password for later use.
+    
+    ![Notepad is open. SQL Connection string is pasted in. {your_password} placeholder is highlighted.](media/sql-connection-string-password-replace.png)
 
-   ![Dynamic Data Masking is highlighted in the left-hand menu.](media/e2-01.png "Dynamic Data Masking")
+4. Go back to the resource list, navigate to your `partsunlimited-web-{uniquesuffix}` **(2)** App Service resource. You can search for `partsunlimited-web` **(1)** to find your app service.
```
Comment (johrose) [inline Hands-on lab/HOL step-by-step - App modernization.md:565]: "You can search for 'partsunlimited-web' **(1)** to find your Web App and App Service Plan"

File: Hands-on lab/HOL step-by-step - App modernization.md [hunk 5]
```diff
@@ -29,1852 +29,1088 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
+16. Go back to Github and observe the repository with the source code uploaded.
 
-In this task, you publish the `Contoso.Web` application into an Azure Web App.
+    ![Github shown with the partsunlimited repository populated with source code.](media/github-partsunlimited-repo-loaded.png)
 
-1. In Visual Studio on your LabVM, right-click the `Contoso.Web` project in the Solution Explorer, and then select **Publish** from the context menu.
+### Task 2: Creating a Staging Deployment Slot
 
-   ![Publish in highlighted in the context menu for the Contoso.Web project.](media/vs-web-publish.png "Publish")
+1. Go back to your lab resource group, navigate to your `partsunlimited-web-{uniquesuffix}` **(2)** App Service resource. You can search for `partsunlimited-web` **(1)** to find your app service.
```
Comment (johrose) [inline Hands-on lab/HOL step-by-step - App modernization.md:688]: "You can search for 'partsunlimited-web' (1) to find your Web App and App Service Plan"

File: Hands-on lab/HOL step-by-step - App modernization.md [hunk 6]
```diff
@@ -29,1852 +29,1088 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
 
-2. On the **Publish** dialog, select **Azure** in the Target box and select **Next**.
+   ![The search box for resources is filled in with partsunlimited-web. The partsunlimited-web-20 Azure App Service is highlighted in the list of resources in the hands-on-lab-SUFFIX resource group.](media/resource-group-appservice-resource.png "Resources")
+   
+2. Switch to the **Deployment slots (1)** tab and select **Add Slot (2)**.
 
-   ![In the Publish dialog, Azure is selected and highlighted in the Target box. The Next button is highlighted.](media/vs-publish-to-azure.png "Publish Web App to Azure")
+    ![App Service Deployment Slots tab is open. Add slot button highlighted.](media/app-service-add-deployment-slot.png)
+    
+3. Type in **staging** as the name **(1)** of the new slot. Select your app service name from the **Clone settings from (2)** dropdown list. This will ensure our web site configurations for the production environment are copied over to the staging environment as a starting point. Select **Add (3)** to add the new slot.
```
Comment (johrose) [inline Hands-on lab/HOL step-by-step - App modernization.md:696]: Would this be select your Web App from the clone settings

File: Hands-on lab/HOL step-by-step - App modernization.md [hunk 7]
```diff
@@ -29,1852 +29,1088 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
 
-3. Next, in the **Specific target** box, select **Azure App Service (Windows)**.
+    ![Add a slot panel is open. Name is set staging. Partsunlimited-web-20 is selected for the clone settings from dropdown list. Add button is highlighted.](media/app-service-staging-slot.png)
+    
+4. Once you receive the success message, close **(1)** the panel. Observe **(2)** the two environments we have for the app service in the deployment slots list.
```
Comment (johrose) [inline Hands-on lab/HOL step-by-step - App modernization.md:700]: Observe the two environments we now have for our Web App in the deployment slots list.

File: Hands-on lab/HOL step-by-step - App modernization.md [hunk 8]
```diff
@@ -29,1852 +29,1088 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
-   ![In the Publish dialog, The Contoso Web App is selected and highlighted under the hands-on-lab-SUFFIX resource group.](media/vs-publish-web-app-service.png "Publish Web App to Azure")
+    ![Deployment slots are listed. Staging slot named partsunlimited-web-20-staging is highlighted.](media/app-service-staging-select.png)
 
-5. Select **Finish**.
+2. Switch to the **Deployment Center (1)** tab. Select **Go to Settings (2)**.
 
-6. Back on the Visual Studio Publish page for the `Contoso.Web` project, select **Publish** to start the process of publishing your Web API to your Azure API App.
+    ![Deployment Center tab is selected. Go to Settings button is highlighted.](media/app-service-goto-deployment-settings.png)
+    
+3. Select **Github (1)** as your source; **.NET Core (2)** as the runtime stack and **.NET Core 2.1 (LTS) (3)** for version. Select **Authorize** to create the connection between the App Service slot and the Github repository we previously prepared.
```
Comment (johrose) [inline Hands-on lab/HOL step-by-step - App modernization.md:714]: between the App Service **deployment** slot and the...

File: Hands-on lab/HOL step-by-step - App modernization.md [hunk 9]
```diff
@@ -29,1852 +29,1088 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
 
-   ![The Publish button is highlighted next to the newly created publish profile on the Publish page.](media/visual-studio-publish-web.png "Publish")
+    ![Deployment Settings page is open. Source is set to Github. Runtime stack is set to .NET Core. Version is set to .NET Core 2.1 (LTS). Authorize button for Github is highlighted. ](media/app-service-deployment-settings.png)
+    
+4. Login with your Github credentials and provide authorization to AppService to access the repository by selecting **Authorize AzureAppService**.
```
Comment (johrose) [inline Hands-on lab/HOL step-by-step - App modernization.md:718]: App Service is two words

File: Hands-on lab/HOL step-by-step - App modernization.md [hunk 10]
```diff
@@ -29,1852 +29,1088 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
+5. Once Github authorization is complete go back to the browser with the Azure Portal. Select the Github **Organization (1)** where you created the Github repository. This might be your personal account name if that is where you created the repository. Select the repository **partsunlimited (2)** and the branch **main (3)** as the source for the CI/CD pipeline. Select **Save (4)** to create CI/CD pipeline.
 
-8. A web browser should open to the published site. If not, open the URL of the published Web App in a browser window.
+    ![Authorize AzureAppService button is highlighted.](media/app-service-cicd-settings-save.png)
 
-9. In the PolicyConnect web page, enter the following credentials to log in, and then select **Log in**:
+Once you select **Save**, the portal will add your app service publishing profile as a secret to your Github repository. This will allow Github Actions to publish the Parts Unlimited web site to the staging deployment slot. Additionally, the portal will create a YAML file that describes the steps required to build and publish the code in the partsunlimited repository.
```
Comment (johrose) [inline Hands-on lab/HOL step-by-step - App modernization.md:726]: App Service is capitalized

File: Hands-on lab/HOL step-by-step - App modernization.md [hunk 11]
```diff
@@ -29,1852 +29,1088 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
+14. Once all changes are complete select **Start commit (4)**. Type a commit message **(5)**. Select **Commit changes (6)** to submit your changes to the repository.
 
-   ![The Storage Account resource is highlighted in the list of resources.](media/resource-group-resources-storage-account.png "Storage account")
+15. Select **Actions (1)** to switch to the workflows page. Notice the latest successfull run **(2)** of our workflow. 
 
-2. From the Storage account Overview blade, select **Containers** under services.
+    ![Actions on the Github Repository is selected. The latest successful run of the workflow is highlighted.](media/github-actions-success.png)
 
-   ![Containers is selected on the Overview blade of the Storage account.](media/storage-account-containers.png "Storage account")
+16. Go back to your lab resource group on the Azure Portal, navigate to your `staging (partsunlimited-web-{uniquesuffix}/staging)` **(2)** App Service resource. You can search for `staging` **(1)** to find your staging app service deployment slot.
```
Comment (johrose) [inline Hands-on lab/HOL step-by-step - App modernization.md:808]: "You can search for 'partsunlimited-web' (1) to find your Web App and App Service Plan"

File: Hands-on lab/HOL step-by-step - App modernization.md [hunk 12]
```diff
@@ -29,1852 +29,1088 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
+    ![The parts{uniquesuffix} Storage Account is highlighted in the list of resources in the hands-on-lab-SUFFIX resource group.](media/select-storage-account.png)
 
-   ![The Contoso.FunctionApp project is expanded and the PolicyDocsFunction.cs file is highlighted and selected.](media/vs-function-app-function.png "Solution Explorer")
+2. Switch to the **Access keys (1)** blade, and select **Show keys (2)**. Select the copy button for the first connection string **(3)**. Paste the value into a text editor, such as Notepad.exe, for later reference.
 
-2. In the `PolicyDocsFunction.cs` file, locate the `TODO #3` block (begins on line 14).
+    ![Access keys blade is open. Show keys button is highlighted. The copy button for the first connection string is pointed.](media/storage-account-connection-copy.png)
 
-   ![The TODO #3 block is highlighted within the PolicyDocsFunction.cs code.](media/vs-policydocsfunction-cs-todo-3.png "PolicyDocsFunction.cs")
+3. Go back to the resource list and navigate to your `partsunlimited-web-{uniquesuffix}` **(2)** App Service resource. You can search for `partsunlimited-web` **(1)** to find your app service.
```
Comment (johrose) [inline Hands-on lab/HOL step-by-step - App modernization.md:944]: "You can search for 'partsunlimited-web' (1) to find your Web App and App Service Plan"

File: Hands-on lab/HOL step-by-step - App modernization.md [hunk 13]
```diff
@@ -29,1852 +29,1088 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
 
-1. In the Visual Studio Solution Explorer, right-click the `Contoso.FunctionApp` project, and then select **Debug** and **Start new instance**.
+### Task 3: Testing Serverless Order Processing
 
-   ![In the Visual Studio Solution Explorer, the context menu for the Contoso.FunctionApp project is displayed, with Debug selected and Start new instance highlighted.](media/vs-function-app-debug.png "Solution Explorer")
+In this task, we will submit a new order on the Parts Unlimited website and observe the order's processing on the order details page. Once the order is submitted, the web front-end will put a job into an Azure Storage Queue. The Function App that we previously deployed is set to listen to the queue and pull jobs for processing. Once order processing is done, a PDF file will be created, and the link for the PDF file will be accessible on the order details page.
 
-2. If prompted, allow the function app to access your local machine resources.
+1. Go back to the resource list and navigate to your `partsunlimited-web-{uniquesuffix}` **(2)** App Service resource. You can search for `partsunlimited-web` **(1)** to find your app service.
```
Comment (johrose) [inline Hands-on lab/HOL step-by-step - App modernization.md:992]: "You can search for 'partsunlimited-web' (1) to find your Web App and App Service Plan"

File: Hands-on lab/HOL step-by-step - App modernization.md [hunk 14]
```diff
@@ -29,1852 +29,1088 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
 
-3. A new console dialog appears, and the function app is loaded. At the of the console, note the output, which provides the local URL of the Function.
+   ![The search box for the resource is filled in with partsunlimited-web. The partsunlimited-web-20 Azure App Service is highlighted in the list of resources in the hands-on-lab-SUFFIX resource group.](media/resource-group-appservice-resource.png "Resources")
 
-   ![The Function console window is displayed with the `PolicyDocs` local URL highlighted.](media/vs-function-app-debug-console.png "Debug")
+2. Select **URL** and navigate to the Parts Unlimited web site hosted in your Azure App Service.
```
Comment (johrose) [inline Hands-on lab/HOL step-by-step - App modernization.md:996]: "hosted in Azure App Service"

File: Whiteboard design session/WDS trainer guide - App modernization.md [hunk 15]
```diff
@@ -190,71 +189,52 @@ Directions: With all participants in the session, the facilitator/SME presents a
...
 
-Contoso employees rely on email as a workflow engine relative to the document management tasks. One group is responsible for scanning and cataloging while another group is responsible for assigning the policies to the specified broker. Manually written emails are sent to brokers when their customers' policy documents have been scanned and indexed. They are using Office 365. The company executives have frequent challenges in gauging productivity and throughput, given the manual workflow. They feel that they are blocked in quickly getting to the insights they need because each new question seems to need more custom development.
+Parts Unlimited has plans to increase its marketing investment, currently on hold because of scaling issues. The company is stuck and cannot grow without increasing its infrastructure footprint. Allegra wants to finalize their cloud vs. on-premises decision based on the current migration effort's success. CFO Jára Cimrman says "We have to drive and scale our e-Commerce presence forward while controlling costs."
```
Comment (johrose) [inline Whiteboard design session/WDS trainer guide - App modernization.md:207]: Who is Allegra?

File: Whiteboard design session/WDS trainer guide - App modernization.md [hunk 16]
```diff
@@ -190,71 +189,52 @@ Directions: With all participants in the session, the facilitator/SME presents a
...
-1. We have seen services like IFTTT (If This, Then That) that let business users automate processes. Does Microsoft Azure offer something similar?
-
-2. Our developers have heard of Logic Apps. Will Microsoft Flow replace these?
+1. Our developers were not able to migrate our .NET Core 2.2 application to a newer version. Should we expect a steep upgrade path with every new version? 
 
-3. Is there a way to securely store application secrets in the cloud?
+2. When a .NET Core version is EoL (End-of-life), does that mean we cannot host our solution in Azure?
 
-4. We noticed that Azure SQL Database does not support all the features available in SQL Server. We are not using these features currently, but are curious to know what the options are for these in Azure? Specifically, we were thinking about Linked Servers, Database Mail, SQL Server Agent Jobs, and Service Broker.
+3. Our engineers love Github. It looks like you bought it. Are you planning to merge it with Azure DevOps?
```
Comment (johrose) [inline Whiteboard design session/WDS trainer guide - App modernization.md:233]: I would amend to "are you planning to integrate with Azure services?"

File: Whiteboard design session/WDS trainer guide - App modernization.md [hunk 17]
```diff
@@ -278,39 +258,35 @@ Directions: With all participants at your table, respond to the following questi
...
+3. What tools would you recommend Parts Unlimited use to migrate its database? How would you use these? Be specific.
 
-5. How would you secure sensitive information used by the website and APIs? Be specific on the Azure Service used, how you would configure it, and how the web or API logic would retrieve its secrets at run time.
+4. What Azure service would host the website?
 
-_Data management_
+5. Would there be any problems with the .NET Core version being EoL (End-of-Life)?
 
-1. What tools would you recommend Contoso use to migrate its database? How would you use these? Be specific.
+_Mondernization_
```
Comment (johrose) [inline Whiteboard design session/WDS trainer guide - App modernization.md:275]: Modernization mis-spelled. Also do we want a question here that alludes to the capabilities of slots given the time we devote to that feature in the HOL?

File: Whiteboard design session/WDS trainer guide - App modernization.md [hunk 18]
```diff
@@ -418,61 +389,47 @@ Directions: Tables reconvene with the larger group to hear the facilitator/SME s
...
 
-_Mobile and Web Applications_
+   ![Architecture diagram of the preferred solution. Azure Functions is implemented to handle order processing. Azure Storage Queue is used to hold the jobs between the web front end and Azure Functions. Github is used to host source code, and Github Actions is used to run the CI/CD.](media/architecture-diagram.png "Preferred solution architecture")
 
-1. How should Contoso implement the PolicyConnect mobile app?
+The solution begins with assessing Parts Unlimited's E-Commerce application with App Service Migration Assistant. Once the assessment succeeds, Parts Unlimited can move to migrate their e-commerce site to an App Service with the App Service Migration Assistant's help. In the case of Parts Unlimited, despite their fears of using an EoL (End-of-Live) .NET Core version, the assessment does not show any blockers. After having App Service Migration Assistant provision an App Service and migrating their application Parts, Unlimited is happy having the front-end moved to the cloud.
```
Comment (johrose) [inline Whiteboard design session/WDS trainer guide - App modernization.md:406]: Parts Unlimited can migrate their e-commerce site to App Service (reword second sentence)

Comment (johrose) [inline Whiteboard design session/WDS trainer guide - App modernization.md:406]: end of life*

Comment (johrose) [inline Whiteboard design session/WDS trainer guide - App modernization.md:406]: *After using the App Service Migration Assistant to provision an App Service plan and migrate their application, Parts Unlimited is happy having the front-end moved to the cloud* (several grammatical issues resolved - would replace final sentence in this paragraph with this)

File: Whiteboard design session/WDS trainer guide - App modernization.md [hunk 19]
```diff
@@ -418,61 +389,47 @@ Directions: Tables reconvene with the larger group to hear the facilitator/SME s
...
+2. What tools would you recommend Parts Unlimited use to assess and migrate its web application? How would you use these? 
 
-_Data management_
+   The App Service Migration Assistant can be used to assess whether their apps have dependencies on unsupported features on Azure App Service. App Service Migration Assistant can run readiness checks and get potential remediation steps for common issues. The assistant provides step-by-step guidance for moving a web app to App Service. Parts Unlimited can use Azure Migrate to consolidate all assessment and migration projects.
```
Comment (johrose) [inline Whiteboard design session/WDS trainer guide - App modernization.md:424]: Would add an intro sentence *Azure App Service is the best solution for running ASP.NET web apps on Azure*

File: Whiteboard design session/WDS trainer guide - App modernization.md [hunk 20]
```diff
@@ -508,109 +465,76 @@ _Data management_
...
+   An App Service.
```
Comment (johrose) [inline Whiteboard design session/WDS trainer guide - App modernization.md:476]: Change to "Azure App Service"

File: Whiteboard design session/WDS trainer guide - App modernization.md [hunk 21]
```diff
@@ -508,109 +465,76 @@ _Data management_
...
 
-_Search_
+5. Would there be any problems with the .NET Core version being EoL (End-of-Life)?
 
-1. Given the Contoso requirements, how would you enable full-text search on the stored policy documents?
+   No, App Service Assessment indicates no issues. However, it is strongly suggested to move to a supported version post-migration. [The .NET Portability Analyzer](https://docs.microsoft.com/en-us/dotnet/standard/analyzers/portability-analyzer) can be used to assess how much work is required for a migration.
```
Comment (johrose) [inline Whiteboard design session/WDS trainer guide - App modernization.md:480]: No, the App Service Migration Assistant indicates....

File: Whiteboard design session/WDS trainer guide - App modernization.md [hunk 22]
```diff
@@ -508,109 +465,76 @@ _Data management_
...
 
-   Contoso could use Azure Cognitive Search and the Indexer for Azure Storage Blobs. For files stored in Azure Storage Blobs, this indexer will full-text index PDFs, Microsoft Office Documents (DOCX, DOC, XLSX, XLS, PPTX, PPT, and MSG), HTML, XML, ZIP, EML, JSON, and plain text files. Contoso could then use the Azure Cognitive Search .NET SDK from the API App to kick off searches.
+   Here is [the Excel report of Portability Analyzer](media/ApiPortAnalysis.xlsx) and the DGML file showing dependencies in Parts Unlimited web application. 
 
-2. How can Azure Cognitive Search be used to extract more information from Contoso's policy documents?
+   ![A Directed Graph Markup Language file that shows the dependency findings of Portability Analyzer](media/portability-analyzer.png)
 
-   Knowledge mining could be used within Azure Cognitive Search to enrich the metadata added to the search index.
+_Mondernization_
```
Comment (johrose) [inline Whiteboard design session/WDS trainer guide - App modernization.md:486]: Modernization misspelled

File: Whiteboard design session/WDS trainer guide - App modernization.md [hunk 23]
```diff
@@ -508,109 +465,76 @@ _Data management_
...
 
-2. Given your answer to the previous question, how would a Contoso business user implement the scenario where a high-priority email is sent to the user's Office 365 email, and in response, an application notification appears on this user's device?
-
-   This could be implemented with Microsoft Flow. The business user would log in to Microsoft Flow and create a new flow. It would begin with the Office 365---on new email, where the priority flag is set to High. Then a Push notification---send a push notification action would follow that can use the subject and sender of the email in the subject of the notification. The user would save this flow. To receive notifications, the user would need to install the Microsoft Flow app on a mobile device.
+   GitHub Actions is an excellent place to automate, customize, and execute your software development workflows right in your repository with. Developers can build, test, and deploy their code right from GitHub. App Services come with a Deployment Center functionality that streamlines Github integration. Parts Unlimited can use the Deployment Center to create a CI/CD pipeline targeting a staging App Service Deployment Slot.
```
Comment (johrose) [inline Whiteboard design session/WDS trainer guide - App modernization.md:512]: App Service does not need to be pluralized in the third sentence

File: Whiteboard design session/WDS trainer guide - App modernization.md [hunk 24]
```diff
@@ -508,109 +465,76 @@ _Data management_
...
+   After the End-Of-Lime time, .NET Core patch updates will no longer be available for .NET Core 2.2. Your application will still run. .NET Core 2.2 was released as a non-LTS (Long Term Support) release. A non-LTS ("Current") release is supported for three months after the next release. For hosting applications that you do not intend to update often, we suggest LTS releases. LTS releases include features and components that have been stabilized, requiring few updates over a longer support release lifetime. The supported upgrade path from .NET Core 2.2 is via .NET Core 3.1. .NET Core 3.1 is released December 3, 2019, as a long-term support release.
 
-   _.NET Core_
+3. Our engineers love Github. It looks like you bought it. Are you planning to merge it with Azure DevOps?
 
-   .NET Core is a cross-platform version of .NET that can run on Windows, Mac, and Linux. This should be used when you are building applications that need to run on any platform, including ASP.NET Core (cross-platform web applications).
+   Both offerings are alive and well. We are happy to offer different options to our customers. You can use both Github and Azure DevOps with your Azure environment.
```
Comment (johrose) [inline Whiteboard design session/WDS trainer guide - App modernization.md:526]: We don't really address ADO at all in the HOL or session, would reframe the question - are you planning to integrate with Azure services, and focus on the GitHub actions capabilities. That or I'd just remove this objection all together. The tension is not something we want to draw attention to

File: Whiteboard design session/WDS trainer guide - App modernization.md [hunk 25]
```diff
@@ -508,109 +465,76 @@ _Data management_
...
+3. Our engineers love Github. It looks like you bought it. Are you planning to merge it with Azure DevOps?
 
-   .NET Core is a cross-platform version of .NET that can run on Windows, Mac, and Linux. This should be used when you are building applications that need to run on any platform, including ASP.NET Core (cross-platform web applications).
+   Both offerings are alive and well. We are happy to offer different options to our customers. You can use both Github and Azure DevOps with your Azure environment.  
 
-   > .NET Framework and .NET Core are two different implementations of the .NET runtime, but they have different profiles of the many APIs and assemblies Microsoft has created for .NET.
+4. We hear a lot about Kubernetes. What is the difference between App Services and Kubernetes? 
 
-   _.NET Standard_
+   Azure is an open cloud that offers multiple choices. We suggest App Service if you have a single or a few back-end services in an N-Tier architecture. Azure Kubernetes Services is a better fit for Cloud-Native (Microservices) design.
```
Comment (johrose) [inline Whiteboard design session/WDS trainer guide - App modernization.md:530]: We suggest App Service for web apps or back-end services in an N-Tier architecture.

**Comment by mimonj** [comment]
Competed the review, looks good.

**Comment by sheltoG** [comment]
Completed the review, it looks good!

File: Whiteboard design session/WDS trainer guide - App modernization.md [hunk 26]
```diff
@@ -508,109 +465,73 @@ _Data management_
...
-
-   The managed instance model allows existing SQL Server customers to lift-and-shift their on-premises applications to the cloud with minimal application and database changes. As a deployment option of Azure SQL Database, it preserves all the PaaS capabilities that drastically reduce management overhead and TCO. While the majority of on-premises SQL features are supported, there are some features, like Filestream, that are not available. Unlike SQL DB, SQL MI does support distributed transactions, SQL Agent jobs, .NET CLR integration, and Service broker.
-
-5. We have been using .NET Framework for years, and now in Visual Studio web have options for .NET Framework, .NET Standard, and .NET Core. As we look at creating our new web and API applications, how do we choose the correct framework?
-
-   _.NET Framework_
+1. Our developers were not able to migrate our .NET Core 2.2 application to a more recent version. Should we expect a steep upgrade path with every new version?
```
Comment (DawnmarieDesJardins) [inline Whiteboard design session/WDS trainer guide - App modernization.md:517]: We're missing a customer objection (#3) from the student guide section.  Student guide has 5 objections, you've answered 4 here.

Comment (daronyondem) [reply Whiteboard design session/WDS trainer guide - App modernization.md:515]: Correct. I have been asked to remove it by @johrose. It looks like I forget to update the student guide. Thanks for catching it. I fixed it.

**Pull request merged**